File: 45주차/전화번호목록.js
function solution(phone_book) {
  var answer = true;
  var dict = {};
  for (let s of phone_book) dict[s] = 1;

  for (let s of phone_book) {
    let tmp = "";
    for (let a of s) {
      tmp += a;
      if (tmp in dict && tmp != s) return false;
    }
  }
  return answer;
}
// 플그머 해시문제

File: 66주차/전화번호목록.py
def solution(phone_book):
    answer = True
    dict = {}
    for v in phone_book:
        dict[v] = 1
    for v in phone_book:
        tmp = ''
        for s in v:
            tmp += s
            if tmp in dict and v != tmp:# 접두사가 번호로 존재한다면 체크. 단, 문자열이 일치하면 접두사가 아님에 조심.
                return False
    return answer 